### PR TITLE
MM-52712: Prevent CI cancellation in master (round 2)

### PR DESCRIPTION
**Pull request by agnivade**
We discovered that cancel-in-progress only controls
in-progress jobs. Which means that pending jobs will _always_
be cancelled regardless. There is an open discussion:
https://github.com/orgs/community/discussions/5435
which was closed saying this is how the feature is designed.

We try to work around this by refactoring each workflow to a template file and calling
them separately from different files for master and PR.

https://mattermost.atlassian.net/browse/MM-52712

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/server-ci-master.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Server CI
```
Comment (isacikgoz) [inline .github/workflows/server-ci-master.yml:1]: I think we should use a unique naming on the workflows as it quickly become conusing on the Ations UI:
resim

**Review by isacikgoz — COMMENTED** [review]
Looks good, I think this would be the silver bullet 🤞 One request below on the naming convention.

**Comment by agnivade** [comment]
Done

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks!

**Review by toninis — APPROVED** [review]
LGTM. I do understand though why you do not want to cancel in progress master jobs ? In any case you want the latest master artifact .

**Comment by agnivade** [comment]
> I do understand though why you do not want to cancel in progress master jobs ? In any case you want the latest master artifact.

Primarily because we run `-race` tests in master only and we want to detect flaky tests. So we want a full CI run on master every time.

Plus, every time a workflow gets cancelled, it shows up as a red cross, so one has to manually click the failure every time to understand whether it's from a cancellation or there's a real failure. More manual work.

**Pull request merged**